facilitator/src/gcp_oauth.rs

```diff
@@ -262,29 +292,32 @@ impl OauthTokenProvider {
         // https://cloud.google.com/iam/docs/reference/credentials/rest/v1/projects.serviceAccounts/generateAccessToken
         let request_url = format!("https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/{}:generateAccessToken",
             service_account_to_impersonate);
-        let request = ureq::post(&request_url)
-            .set(
-                "Authorization",
-                &format!("Bearer {}", self.ensure_default_account_token()?),
-            )
-            .build();
-        let http_response = send_json_request(JsonRequestParameters {
-            request: request,
-            body: ureq::json!({
+
+        let request = prepare_request(
+            Some(self.agent.clone()),
+            RequestParameters {
+                url: Url::parse(&request_url).context(format!(
+                    "failed to parse request_url to generate access token: {}",
+                    &request_url
+                ))?,
+                method: Method::Post,
+                token_provider: Some(self),
```

Comment:
I don't think this will work. `prepare_request` calls `RequestParameters::authenticated_request` which calls `token_provider.ensure_oauth_token` which will call back into this and you will have an infinite recursion. I think you just need to call `self.ensure_default_account_token()` and wrap that in a `StaticOauthTokenProvider`.